src/app/dao/tests/JoinerDAO.spec.ts

```diff
@@ -1,17 +1,462 @@
+
+                    // Then it should fail
+                    await expectPermissionToBeDenied(result);
+                }
+            });
+        });
+    });
+    describe('for non-chosen candidate', () => {
+        it('should allow to remove themself from the candidates', async() => {
+            // Given a part and joiner, and a candidate
+            const partId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
+            const candidate: MinimalUser = await addCandidate(partId, false);
+
+            // When the candidate removes themself from the list
+            const result: Promise<void> = joinerDAO.removeCandidate(partId, candidate);
+
+            // Then it should succeed
+            await expectAsync(result).toBeResolvedTo();
+        });
+        it('should forbid changing its own candidate fields', async() => {
+            // Given a part and joiner, with a candidate
+            const partId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
+            const candidate: MinimalUser = await addCandidate(partId);
+
+            await createConnectedUser(MALICIOUS_EMAIL, MALICIOUS_NAME);
+
+            // When the malicious users tries to change one of its own candidate's fields
+            const update: Partial<MinimalUser> = { name: 'foo' };
+            const result: Promise<void> = joinerDAO.subCollectionDAO(partId, 'candidates').set(candidate.id, update);
+
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+        it('should forbid to change fields', async() => {
+            // Given a part, and a candidate (that is not the creator)
+            const partId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
+            const candidate: MinimalUser = await addCandidate(partId, false);
+
+            const updates: Partial<Joiner>[] = [
+                { chosenOpponent: candidate },
+                { partStatus: PartStatus.CONFIG_PROPOSED.value },
+                { firstPlayer: FirstPlayer.CHOSEN_PLAYER.value },
+                { partType: PartType.BLITZ.value },
+                { maximalMoveDuration: 73 },
+                { totalPartDuration: 1001001 },
+            ];
+            for (const update of updates) {
+                // When modifying a field
+                const result: Promise<void> = joinerDAO.update(partId, update);
+
+                // Then it should fail
+                await expectPermissionToBeDenied(result);
+            }
+        });
+    });
+    describe('for chosen opponent', () => {
+        let partId: string;
+        beforeEach(async() => {
+            // Given a joiner where the current user is set as chosen opponent
+
+            // We have to follow a realistic workflow to achieve that
+            // The part is first created
+            const createdPartId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
```

Comment:
here too, why the undeeded temp var ?

Reply:
oubli de refactor j'imagine, c'est fixed nu